crates/holochain/build.rs

```diff
@@ -119,6 +119,32 @@ mod version_info {
     }
 }
 
+mod schema_definitions {
```

Comment:
Added this to the `holochain` build script, so that building `holochain` will generate the conductor config schema file that gets stored in the `holochain_conductor_api` crate. Open to suggestions to better places to have this.

Reply:
I'd wait for others to confirm this makes sense, but my recommendation would be to move this into the build script for `holochain_conductor_api` and write the schema file in the build output directory for that crate.

Reply:
I intended to have it inside `holochain_conductor_api` but it's not possible to have `holochain_conductor_api` be a dependency inside it's own build script

Reply:
ah right -- does it make sense to make a new crate `holochain_conductor_api_config_schema` just for building the schema? I think ideally we would generate the schema at build time, and shouldn't need to include `holochain` as a dependency just to get the schema for `holochain_conductor_api`

Reply:
The conductor config is used to configure a conductor, whereas the conductor API is a websocket interface for making requests to a running conductor. It seems that the config has got nothing to do with the API. I suppose the only reason for it being placed here is so that it's not in the huge holochain crate and can be exposed. By default I'd expect it to be near the conductor.

This is not the time to break the conductor out of holochain, so perhaps a separate crate for the config is in order?

Reply:
I anticipate we'll need JSON schema generation for other configurations beyond just the conductor config. I understand the suggestion of creating a json_schema crate, but I think we should clarify its intended functionality:

- What API would this crate expose beyond build script functionality?
- Would it be a library crate with runtime schema generation capabilities?
- Or would it be more of a CLI tool for generating schemas?

Reply:
I think @jost-s proposal is that we extract the config structs out of `holochain_conductor_api` to a new crate `holochqin_conductor_config` so that the _api crate is just for the websocket interfaces. I'm agnostic on this.

My proposal is to add a `holochain_conductor_api_schema` (or if we went with jost's proposal, then `holochain_conductor_config_schema`). This crate would be empty except for building the json schema for that one specific crate.

If we need to generate schemas for other crates in the future I'd propose a separate _schema crate for each, rather than a single `json_schema` crate for all, to avoid the unnecessary dependencies. (I.e. we only need a schema for the conductor config in this case  so no need to depend on a bunch of other crates that we might also be generating schemas for).